client/clientimpl_test.go

```diff
@@ -1384,11 +1385,15 @@ func verifyUpdatePackages(t *testing.T, testCase packageTestCase) {
 			// Wait until all syncing is done.
 			<-syncerDoneCh
 
-			for pkgName, receivedContent := range localPackageState.GetContent() {
+			// Clone localPackageState instead of using directly to avoid a race when testing for the DOWNLADING status
```

Comment:
The previous line supposedly wait until all synching is done. Doesn't that mean status can no longer be modified after the previous line and that localPackageState will not change anymore? What code is doing a concurrent write that requires cloning the state? And if there is code that is doing concurrent write, isn't Clone() operation itself under the risk of racing with that?

Reply:
I think what's occurring is that `<-syncerDonCh` gets unblocked when the expected status is found; if this is set to `Downloading`, the package sync continues in the background. I'll try to reqork the test to avoid cloning